File: monsoon3.0/src/main/java/wtf/monsoon/api/util/misc/PacketSleepThread.java
package wtf.monsoon.api.util.misc;

import net.minecraft.network.Packet;

public class PacketSleepThread extends Thread {

    private final Packet<?> packet;
    private final long delay;

    public PacketSleepThread(Packet<?> packet, long delay) {
        this.packet = packet;
        this.delay = delay;
    }

    @Override
    public void run() {
        try {
            sleep(this.delay);
            PacketUtil.sendPacketNoEvent(packet);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/util/network/PacketSleepThread.kt
package wtf.monsoon.client.util.network

import net.minecraft.network.Packet

class PacketSleepThread(private val packet: Packet<*>, private val delay: Long) : Thread() {
    override fun run() {
        try {
            sleep(delay)
            PacketUtil.sendPacketNoEvent(packet)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}